app/streaming/input.cpp

```diff
@@ -884,30 +895,42 @@ void SdlInputHandler::rumble(unsigned short controllerNumber, unsigned short low
 
     // Stop the last effect we played
     if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
-        SDL_HapticDestroyEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId);
+        if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_LEFTRIGHT) {
+            SDL_HapticDestroyEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId);
+        } else if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_SIMPLERUMBLE) {
+            SDL_HapticRumbleStop(haptic);
+        }
     }
 
     // If this callback is telling us to stop both motors, don't bother queuing a new effect
     if (lowFreqMotor == 0 && highFreqMotor == 0) {
         return;
     }
 
-    SDL_HapticEffect effect;
-    SDL_memset(&effect, 0, sizeof(effect));
-    effect.type = SDL_HAPTIC_LEFTRIGHT;
+    if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_LEFTRIGHT) {
+        SDL_HapticEffect effect;
+        SDL_memset(&effect, 0, sizeof(effect));
+        effect.type = SDL_HAPTIC_LEFTRIGHT;
 
-    // The effect should last until we are instructed to stop or change it
-    effect.leftright.length = SDL_HAPTIC_INFINITY;
+        // The effect should last until we are instructed to stop or change it
+        effect.leftright.length = SDL_HAPTIC_INFINITY;
 
-    // SDL haptics range from 0-32767 but XInput uses 0-65535, so divide by 2 to correct for SDL's scaling
-    effect.leftright.large_magnitude = lowFreqMotor / 2;
-    effect.leftright.small_magnitude = highFreqMotor / 2;
+        // SDL haptics range from 0-32767 but XInput uses 0-65535, so divide by 2 to correct for SDL's scaling
+        effect.leftright.large_magnitude = lowFreqMotor / 2;
+        effect.leftright.small_magnitude = highFreqMotor / 2;
 
-    // Play the new effect
-    m_GamepadState[controllerNumber].hapticEffectId = SDL_HapticNewEffect(haptic, &effect);
-    if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
-        SDL_HapticRunEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId, 1);
+        // Play the new effect
+        m_GamepadState[controllerNumber].hapticEffectId = SDL_HapticNewEffect(haptic, &effect);
+        if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
+            SDL_HapticRunEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId, 1);
+        }
+    } else if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_SIMPLERUMBLE) {
+        SDL_HapticRumblePlay(haptic,
+                             std::min(1.0, std::max(highFreqMotor, lowFreqMotor)/65535.0),
```

Comment:
I'm not sure that we want `std::max(highFreqMotor, lowFreqMotor)` here.

The low and high frequency motors have different sized weights on them, so they feel different. We should try to mimic that difference as well as we can given the API limitations.

On Android, where I have a similar issue, I scaled them so 75% of the amplitude was the low frequency motor and 25% was the high frequency motor. We should do something like that here (or if you have a better idea, I'm all ears).

Reply:
I try make some tests with different weights — but need some time. Maybe there will be some ideas.

Reply:
Fixed. I've tested some weights for motors and figured out, that hi freq motor is really lightweight. So  I've set 0.33 and 0.8 as weights (defines in input.h file). On my senses, it’s better then 0.25 and 0.75. But you can define that values on your own.

Reply:
Nice, thanks.